### PR TITLE
Rename Hive connector "hive-hadoop2 to "hive"

### PR DESCRIPTION
**Pull request by electrum**
This simplification is long overdue. The original name is kept for compatibility with existing configurations, but a warning is logged when it is used. The plugin directory is also renamed inside the server tarball. We should name the Maven artifact as well, but I'll leave that for a follow up, as it's not obvious what to call it.

### CODE REVIEW

**Comment by electrum** [comment]
cc @mosabua

**Review by dain — APPROVED** [review]

**Review by kokosing — APPROVED** [review]

**Comment by mosabua** [comment]
Wow ...

File: docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -174,13 +174,13 @@ Configuration
 -------------
 
 Create ``etc/catalog/hive.properties`` with the following contents
-to mount the ``hive-hadoop2`` connector as the ``hive`` catalog,
+to mount the ``hive`` connector as the ``hive`` catalog,
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:177]: Are these really the only instances of the connector name in all the docs?

**Comment by findepi** [comment]
from @electrum 's https://github.com/trinodb/trino/pull/5625#pullrequestreview-515006816

> At a minimum, we also need to rename the plugin/hive-hadoop2 directory in the tarball. This will be a breaking change for people that do things like add libraries

are we ok with the breaking change?
cc @trinodb/maintainers

File: docs/src/main/sphinx/connector/hive.rst (hunk 1 above)
Comment (electrum) [reply docs/src/main/sphinx/connector/hive.rst:177]: Yes

**Comment by electrum** [comment]
@findepi sorry, I completely forgot about your PR, though I see you closed it. I'm glad you remembered my comment about this breaking people that install libraries. We should call this out, but I'm not concerned, since that's an "unsupported" activity and needs to be handled carefully anyway.

**Comment by electrum** [comment]
We could go a different way and add a `getLegacyNames()` method to `ConnectorFactory`, similar to `@LegacyConfig` in Airlift config. This might be useful for other connectors like MemSQL, for which this approach wouldn't work. But we can always do that later and migrate the Hive connector to use the new mechanism.

**Pull request merged**